### PR TITLE
Create contributing.md

### PR DESCRIPTION
**Pull request by OwenArnold**
Welcome contributions and set some simple guidelines.

A few example directories have been added. I'm not adding for ALL techniques as there might be discussion about whether we (for example) put all spectroscopy under one root directory, or whether we have `direct`, `indirect` as non-nested directories.

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Contributing
+
+We welcome contributions to this repository. When contributing to this repository, we encourage you to discuss the change you wish to make via issue, email, or any other method with the owners of this repository before making a change.
```
Comment (nvaytet) [inline contributing.md:3]: Can we have one sentence per line if possible?
Makes it easier to edit/review.

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+### Testing
+
+To ensure that the package is robust we are very keen that authors provide unit tests alongside code. It is possible that future updates for `scipp`, `scippneutron` dependencies can break the code you contribute. If we are aware of failing tests, we can provide future fixes and migrations for you. Please avoid large data files, or any code requiring network access. Test suites should be fast to execute.
+
+## Documentation
+
+Please provide and update documentation. Put python docstrings on your user facing functions, provide code comments and consider other explanations you need to provide to desribe how your functions work. We will build and publish sphinx documentation located [here](https://github.com/scipp/ess/tree/main/docs).
```
Comment (nvaytet) [inline contributing.md]:
```suggestion
Please provide and update documentation. Put python docstrings on your user facing functions, provide code comments and consider other explanations you need to include to describe how your functions work. We will build and publish sphinx documentation located [here](https://github.com/scipp/ess/tree/main/docs).
```

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+## Contributing
+
+We welcome contributions to this repository. When contributing to this repository, we encourage you to discuss the change you wish to make via issue, email, or any other method with the owners of this repository before making a change.
+
+Code contributions should be made via a github pull request (PR). Once a PR has been submitted, we conduct a review, so please anticipate some discussion on anything you submit to ensure we can keep the code base of a good quality and maintainable.
+
+### Layout
+
+This repository is ess instrument centric and we therefore strongly prefer having all code organised into `src/ess/{instrument}` directories. If your code is specific to the ess facility, but is intended to be ubiquitous across a class of instruments, you may put it into a technique specfic directory `src/ess/{technique}` i.e `src/ess/reflectometry`. Code that is technique specific, but free from ess facility considerations may be considered to go in [scippneutron](https://github.com/scipp/scippneutron)
```
Comment (SimonHeybrock) [inline contributing.md]: Remember our guideline of one line per sentence, for diff readability.

**Review by nvaytet — CHANGES_REQUESTED** [review]
- I think we need to add all folders: `reflectometry`, etc.
- I would also add instrument folders

**Comment by nvaytet** [comment]
I would vote for one spectroscopy directory

**Comment by OwenArnold** [comment]
> * I think we need to add all folders: `reflectometry`, etc.

As per the pr description, I'm not sure of the best way of organising this for all techniques.

File: contributing.md [hunk 4]
```diff
@@ -2,16 +2,24 @@
...
+Code contributions should be made via a github pull request (PR).
+Once a PR has been submitted, we conduct a review, so please anticipate some discussion on anything you submit to ensure we can keep the code base of a good quality and maintainable.
 
 ### Layout
 
-This repository is ess instrument centric and we therefore strongly prefer having all code organised into `src/ess/{instrument}` directories. If your code is specific to the ess facility, but is intended to be ubiquitous across a class of instruments, you may put it into a technique specfic directory `src/ess/{technique}` i.e `src/ess/reflectometry`. Code that is technique specific, but free from ess facility considerations may be considered to go in [scippneutron](https://github.com/scipp/scippneutron)
+This repository is ess instrument centric and we therefore strongly prefer having all code organised into `src/ess/{instrument}` directories.
+If your code is specific to the ess facility, but is intended to be ubiquitous across a class of instruments, you may put it into a technique specfic directory `src/ess/{technique}` i.e `src/ess/reflectometry`.
```
Comment (nvaytet) [inline contributing.md]:
```suggestion
If your code is specific to the ess facility, but is intended to be ubiquitous across a class of instruments, you may put it into a technique specific directory `src/ess/{technique}` i.e `src/ess/reflectometry`.
```

**Review by nvaytet — COMMENTED** [review]
Missing dirs?
`beer`, `freia`, `bifrost`, `cspec`

**Review by nvaytet — APPROVED** [review]
Looks good now.
We have to be careful with keeping this repo clean, as there will probably be many contributors.
For instance, after this is merged we would have a `imaging` folder, but we also have a `v20/imaging` folder.

**Comment by nvaytet** [comment]
Although, should we not have at least a reflectometry folder so that @arm61 can put his files in the right places?

**Comment by OwenArnold** [comment]
> Although, should we not have at least a reflectometry folder so that @arm61 can put his files in the right places?

We should, I'm avoiding potential conflict by not making it as I know it's coming from his PR.

**Comment by OwenArnold** [comment]
> We have to be careful with keeping this repo clean, as there will probably be many contributors.

Fully agree.

**Pull request merged**